draft-add-requirements.md

```diff
@@ -55,66 +92,119 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 document are to be interpreted as described in BCP 14 {{RFC2119}} {{!RFC8174}}
 when, and only when, they appear in all capitals, as shown here.
 
-# Discovery of associated resolvers {#first-usecase}
-
+{: #first-usecase}
+# Discovery of associated resolvers
 A client may begin with information from the attached networks, and/or resolvers known from configuration.
 This information may be used to then discover one or more associated or equivalent encrypted resolvers.
 These are expected to be operated by the same entity that provides the resolver the client started with, but
-may not have identical behaviour. For example one entity may offer a set of encrypted resolvers with different levels of
-filtering or proximity.
+may not have identical behaviour. For example one entity may offer a set of encrypted resolvers with different levels
+of filtering or proximity.
+
+These resolver upgrade mechanisms can either add new parameters to existing provisioning mechanisms
+(adding necessary information to use DoT or DoH to options in DHCP, RAs, or IKEv2) or else provide a way
+to communicate with a provisioned unencrypted DNS resolver and discover the equivalent or associated encrypted DNS resolver.
 
 | Requirement | Description |
 | R1.1 | There must be a mechanism for a client to learn the set of encrypted resolvers that are associated with a resolver that is known only by its IP address. |
 | R1.2 | Discovery must be possible even when the IP address is only valid locally. |
 | R1.3 | More to be added |
 
-## Discover resolvers associated with one declared by an attached network
+## Network-provisioned resolvers {#local-network}
+
+DNS servers are often provisioned by a network as part of DHCP options {{?RFC2132}},
+IPv6 Router Advertisement (RA) options {{?RFC8106}}, Point-to-Point Protocol (PPP) {{?RFC1877}},
+3GPP Protocol Configuration Options, or another mechanism. Historically this is usually one or
+more DNS resolver IP addresses, to be used for traditional unencrypted DNS. However it could also
+be a richer set of information.
 
-During network attach, networks frequently announce a resolver that they recommend clients should use.
-This could be through DHCP, RA, PPP, PCO, or another mechanism. Historically this is usually a Do53 resolver,
-declared simply by its IP address. However it could also be a richer set of information.
+Using an encrypted resolver that is associated to the one provisioned by the network
+can provide several benefits that are not possible if only unencrypted DNS is used:
+
+- Prevent other devices on the network from observing client DNS messages
+- Verify that answers come from the selected DNS resolver
+- Authenticate that the DNS resolver is the one provisioned by the network
+
+Often, network-provisioned resolvers are forwarders running on a local router. The discovered
```

Comment:
network-provisioned resolvers may not be a forwaders in large enterprise networks. The above statement is only valid for home network deployments.

Reply:
I don't think forwarders are exclusive to homes. They're also in SMEs, and there must be some large enterprises that use forwarders in remote offices. But we could amend "Often" to "Frequently" or "Sometimes".

Reply:
Agreed, small and branch offices typically use forwarders but company HQ for large enterprises typically use recursive servers. "Frequently" works for me.

Reply:
Done.